pkgs/applications/networking/browsers/chromium/common.nix

```
@@ -526,6 +527,17 @@ let
       # preventing compilations of chromium with versions below their intended version, not about running the very
       # exact version or even running a newer version.
       ./patches/chromium-136-nodejs-assert-minimal-version-instead-of-exact-match.patch
+    ]
+    ++ [
+      (
+        if (chromiumVersionAtLeast "138") then
+          (fetchurl {
+            url = "https://src.fedoraproject.org/rpms/chromium/raw/9a0191f60e56b662ac98927bd4788dfa0cd60b0f/f/chromium-139-rust-FTBFS-suppress-warnings.patch";
+            hash = "sha256-ItMR6uSRKIUpQ9AeYTvEVjGQulzS38mHiclRYZmlqNw=";
+          })
+        else
+          ./patches/chromium-137-rust-FTBFS-suppress-warnings.patch
+      )
```

I haven't fully tested this yet, but I believe `-Amismatched_lifetime_syntaxes` is all we need here. That would also allow us to cherry-pick this from upstream instead of some patch from fedora. 

```suggestion
    ]
    ++ lib.optionals (chromiumVersionAtLeast "138") [
      (fetchpatch {
        # Unbreak building with Rust 1.89+ which introduced
        # a new mismatched_lifetime_syntaxes lint.
        # https://issues.chromium.org/issues/424424323
        name = "chromium-138-rust-1.86-mismatched_lifetime_syntaxes.patch";
        # https://chromium-review.googlesource.com/c/chromium/src/+/6658267
        url = "https://chromium.googlesource.com/chromium/src/+/94a87ff38c51fd1a71980a5051d3553978391608^!?format=TEXT";
        decode = "base64 -d";
        includes = [ "build/rust/cargo_crate.gni" ];
        hash = "sha256-xf1Jq5v3InXkiVH0uT7+h1HPwZse5MDcHKuJNjSLR6k=";
      })
    ]
    ++ lib.optionals (!chromiumVersionAtLeast "138") [
      # Rebased variant of the patch above for
      # electron 35 (M134) and 36 (M136)
      ./patches/chromium-134-rust-1.86-mismatched_lifetime_syntaxes.patch
```